src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -666,7 +781,16 @@ class ShenandoahHeap : public CollectedHeap, public ShenandoahSpaceInfo {
   static inline void atomic_clear_oop(narrowOop* addr,       oop compare);
   static inline void atomic_clear_oop(narrowOop* addr, narrowOop compare);
 
-  void trash_humongous_region_at(ShenandoahHeapRegion *r);
+  size_t trash_humongous_region_at(ShenandoahHeapRegion *r);
+
+  static inline void increase_object_age(oop obj, uint additional_age);
+
+  // Return the object's age, or a sentinel value when the age can't
+  // necessarily be determined because of concurrent locking by the
+  // mutator
+  static inline uint get_object_age(oop obj);
```

Comment:
Once https://github.com/openjdk/jdk/pull/20067 is integrated, you could force LW-locking with that new monitor-mapping and enjoy simple hassle-free updating of the age bits in the object header.

Reply:
@ysramakrishna will create a tracking ticket so that we don't forget about this.